daemon/stats_collector.go

```diff
@@ -76,22 +76,43 @@ func (s *statsCollector) unsubscribe(c *Container, ch chan interface{}) {
 }
 
 func (s *statsCollector) run() {
+	type publishersPair struct {
+		container *Container
+		publisher *pubsub.Publisher
+	}
+	// cannot determine capacity of an array here,
+	// it grows enough after the first iteration
+	pairs := make([]publishersPair, 0)
```

Comment:
I'm not sure that `pairs = pairs[:0]` prevent allocations. I think new slice will be anyway created on first append.